### PR TITLE
docs: fix details about WorkerInfo vs TestInfo availability

### LINKED ISSUE #21695: [BUG] wrong type indicated for beforeAll and afterAll hooks
### System info
- Playwright Version: 8.19.1
- Operating System: All
- Browser: All
- Other info:

### Source code

```ts
test.describe('Test', () => {
  // should be WorkerInfo but is typed as TestInfo
  test.beforeAll(async ({ browser }, workerInfo) => {});
  // correctly typed as TestInfo
  test.afterEach(async ({}, testInfo) => {});
```

**Steps**
No special config or steps are needed - you can also simply check the `test.d.ts` type definitions file to confirm this:

```ts
  /**
   * Declares a `beforeAll` hook that is executed once per worker process before all tests.
   *
   * **Details**
   *
   * When called in the scope of a test file, runs before all tests in the file. When called inside a
   * [test.describe(title, callback)](https://playwright.dev/docs/api/class-test#test-describe-1) group, runs before all
   * tests in the group. If multiple `beforeAll` hooks are added, they will run in the order of their registration.
   *
   * Note that worker process is restarted on test failures, and `beforeAll` hook runs again in the new worker. Learn
   * more about [workers and failures](https://playwright.dev/docs/test-retries).
   *
   * You can use [test.afterAll(hookFunction)](https://playwright.dev/docs/api/class-test#test-after-all) to teardown
   * any resources set up in `beforeAll`.
   *
   * **Usage**
   *
   * ```js
   * // example.spec.ts
   * import { test, expect } from '@playwright/test';
   *
   * test.beforeAll(async () => {
   *   console.log('Before tests');
   * });
   *
   * test.afterAll(async () => {
   *   console.log('After tests');
   * });
   *
   * test('my test', async ({ page }) => {
   *   // ...
   * });
   * ```
   *
   * @param hookFunction Hook function that takes one or two arguments: an object with worker fixtures and optional [TestInfo].
   */
  beforeAll(inner: (args: TestArgs & WorkerArgs, testInfo: TestInfo) => Promise<any> | any): void;
  /**
   * Declares an `afterAll` hook that is executed once per worker after all tests.
   *
   * **Details**
   *
   * When called in the scope of a test file, runs after all tests in the file. When called inside a
   * [test.describe(title, callback)](https://playwright.dev/docs/api/class-test#test-describe-1) group, runs after all
   * tests in the group. If multiple `afterAll` hooks are added, they will run in the order of their registration.
   *
   * Note that worker process is restarted on test failures, and `afterAll` hook runs again in the new worker. Learn
   * more about [workers and failures](https://playwright.dev/docs/test-retries).
   *
   * **Usage**
   *
   * ```js
   * test.afterAll(async () => {
   *   console.log('Done with tests');
   *   // ...
   * });
   * ```
   *
   * @param hookFunction Hook function that takes one or two arguments: an object with worker fixtures and optional [TestInfo].
   */
  afterAll(inner: (args: TestArgs & WorkerArgs, testInfo: TestInfo) => Promise<any> | any): void;
```

**Expected**

According [to the documentation](https://playwright.dev/docs/api/class-workerinfo) the WorkerInfo class is available to beforeAll and afterAll hooks as well as worker scoped fixtures.

The docs describing the [beforeAll and afterAll hooks](https://playwright.dev/docs/api/class-test#test-before-all) contradict the WorkerInfo docs. They probably need updating too.

**Actual**

A TestInfo type is returned instead of the correct WorkerInfo type.

### PR DESCRIPTION
**Pull request by dgozman**
Fixes #21695.

### CODE REVIEW

**Review by pavelfeldman — APPROVED** [review]

File: docs/src/test-api/class-testinfo.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 * since: v1.10
 * langs: js
 
-`TestInfo` contains information about currently running test. It is available to any test function, [`method: Test.beforeEach`] and [`method: Test.afterEach`] hooks and test-scoped fixtures. `TestInfo` provides utilities to control test execution: attach files, update test timeout, determine which test is currently running and whether it was retried, etc.
+`TestInfo` contains information about currently running test. It is available to any test function, before/after hook or a test-scoped fixture. `TestInfo` provides utilities to control test execution: attach files, update test timeout, determine which test is currently running and whether it was retried, etc.
```
Comment (GrayedFox) [inline docs/src/test-api/class-testinfo.md:5]: The links are kind of nice for navigating between related hooks/methods etc :bowtie: 

Also at the risk of sounding knit picky - best to be explicit where possible (before/after hooks could be interpreted as beforeAll and afterAll only for people coming from Mocha land). Consider:

```
`TestInfo` contains information about the currently running test. It is available to all test functions, all test hook functions ([`method: Test.beforeAll`], [`method: Test.beforeEach`], [`method: Test.afterEach`], and [`method: Test.afterAll`]) and inside test-scoped fixtures.

`TestInfo` provides utilities to control test execution: attach files, update test timeout, determine which test is currently running and whether it was retried, etc.
```

File: docs/src/test-api/class-workerinfo.md [hunk 2]
```diff
@@ -2,23 +2,8 @@
 * since: v1.10
 * langs: js
 
-`WorkerInfo` contains information about the worker that is running tests. It is available to [`method: Test.beforeAll`] and [`method: Test.afterAll`] hooks and worker-scoped fixtures.
+`WorkerInfo` contains information about the worker that is running tests, available to the worker-scoped fixtures. `WorkerInfo` is a subset of [TestInfo] that is available in many other places.
```
Comment (GrayedFox) [inline docs/src/test-api/class-workerinfo.md:5]: `WorkerInfo` contains information about the worker that is running tests and is available to worker-scoped fixtures. `WorkerInfo` is a subset of [TestInfo] that is available in many other places.

File: packages/playwright-test/types/test.d.ts [hunk 3]
```diff
@@ -1800,19 +1800,8 @@ export interface FullConfig<TestArgs = {}, WorkerArgs = {}> {
...
- *
- * ```js
- * import { test, expect } from '@playwright/test';
- *
- * test.beforeAll(async ({ browserName }, workerInfo) => {
- *   console.log(`Running ${browserName} in worker #${workerInfo.workerIndex}`);
- * });
- * ```
- *
+ * `WorkerInfo` contains information about the worker that is running tests, available to the worker-scoped fixtures.
```
Comment (GrayedFox) [inline packages/playwright-test/types/test.d.ts:1803]: `...and is available to worker-scoped fixtures`

File: packages/playwright-test/types/test.d.ts [hunk 4]
```diff
@@ -1844,11 +1833,9 @@ export interface WorkerInfo {
 }
 
 /**
- * `TestInfo` contains information about currently running test. It is available to any test function,
- * [test.beforeEach(hookFunction)](https://playwright.dev/docs/api/class-test#test-before-each) and
- * [test.afterEach(hookFunction)](https://playwright.dev/docs/api/class-test#test-after-each) hooks and test-scoped
- * fixtures. `TestInfo` provides utilities to control test execution: attach files, update test timeout, determine
- * which test is currently running and whether it was retried, etc.
+ * `TestInfo` contains information about currently running test. It is available to any test function, before/after
```
Comment (GrayedFox) [inline packages/playwright-test/types/test.d.ts:1836]: See above

**Review by GrayedFox — APPROVED** [review]
Style and clarity comments only (annoying human doc linter alert) - so nothing blocking 🍻

**Comment by dgozman** [comment]
> Style and clarity comments only (annoying human doc linter alert) - so nothing blocking 🍻

Any stylistic comments are appreciated 🙏

**Pull request merged**